### DISCUSSION TITLE
HA Core 2022.2.3, errore nel riavvio da GUI

### DISCUSSION THREAD

**Discussion by manny199** [indomus/forum#1472] [GENERALE - Discussioni generiche]
Ciao,
ho appena aggiornato home assistant core alla versione 2022.2.3.
Se tento il classico riavvio da sistema --> gestione del server --> riavvia, HA va in stop ed è necessario un riavvio da terminale tramite reboot. Nessun errore nel log e vedo il database stoppato.
Ho cercato ma non lo trovo come problema noto.
Qualcuno ha incontrato questo malfunzionamento?

**Comment by snakuzzo** [comment]
Guarda i log...se non si riavvia può darsi che ci sia qualcosa che non va. Quando fai il riavvio in automatico fa anche un controllo della configurazione

**Reply by manny199** [reply]
Nessun errore nel log (parlo del log.txt nella directory di configurazione).
Peraltro nel configuration.yaml non ho fatto alcuna modifica.
L'unica cosa fatta è stato l'aggiornamento di HA alla nuova versione

**Reply by snakuzzo** [reply]
Non esiste alcun `log.txt` di HA. Il file di log è `home-assistant.log`
Io ho aggiornato poco fa e non riscontro problemi con il riavvio.

**Reply by manny199** [reply]
@snakuzzo 
Avevo scritto log.txt per abbreviare ...
Non ho errori da nessuna parte e il file `home-assistant.log` è pulito e non contiene neppure warning.
Non appena faccio click su "RIAVVIA", se guardo il database, rimane uno solo dei 3 file, come se fermasse il servizio home assistant.
Il pulsante "RIAVVIA" si comporta come se fosse "FERMA" e mi stoppa il servizio.
Dunque devo avviare manualmente il servizio da riga di comando con `sudo systemctl start home-assistant@homeassistant`
Questa cosa mi fa impazzire

**Reply by snakuzzo** [reply]
Prova a mettere il log a debug con [logger](https://www.home-assistant.io/integrations/logger/).
Comunque, se posso permettermi, ti consiglio di passare a Docker

**Reply by manny199** [reply]
@snakuzzo 
Logger mi crea un file lunghissimo.
Mi pare di aver capito che serve il debug, mi spieghi come estrarre solo il dato necessario?

**Reply by snakuzzo** [reply]
Devi cercare tu, non posso dirtelo io

**Reply by snakuzzo** [reply]
mettiti in tail e vedi cosa succede quando premi il tasto per riavviare

**Reply by manny199** [reply]
Questo quello che leggo quando premo per riavvio:
```
2022-02-07 18:59:31 DEBUG (MainThread) [custom_components.hacs] HacsTask<store_hacs_data> Executing task
2022-02-07 18:59:31 DEBUG (MainThread) [custom_components.hacs] Saving data
2022-02-07 18:59:31 DEBUG (SyncWorker_5) [homeassistant.helpers.storage] Writing data for trace.saved_traces to /home/homeassistant/.homeassistant/.storage/trace.saved_traces
2022-02-07 18:59:31 DEBUG (SyncWorker_3) [homeassistant.helpers.storage] Writing data for core.restore_state to /home/homeassistant/.homeassistant/.storage/core.restore_state
2022-02-07 18:59:31 DEBUG (MainThread) [custom_components.hacs] Did not store data for 'hacs.hacs'. Content did not change
2022-02-07 18:59:31 DEBUG (MainThread) [custom_components.hacs] Did not store data for 'hacs.repositories'. Content did not change
2022-02-07 18:59:31 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event hacs/repository[L]>
2022-02-07 18:59:31 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event hacs/config[L]>
2022-02-07 18:59:31 DEBUG (MainThread) [custom_components.hacs] HacsTask<store_hacs_data> took 0.180 seconds to complete
2022-02-07 18:59:31 DEBUG (MainThread) [homeassistant.core] Bus:Handling <Event homeassistant_close[L]>`
```

**Reply by snakuzzo** [reply]
Da qui non mi pare ci siano problemi

**Reply by manny199** [reply]
@snakuzzo Tieni presente che sino alla versione 2021.12 tutto funzionava perfettamente.
Sono dunque portato a pensare che ci sia qualche componente non più supportato che crea problemi di questo genere.
Non so più dove e cosa guardare e reinstallare tutto mi rode tantissimo

**Reply by snakuzzo** [reply]
Se fosse come dici lo vedresti nel log.
Prova a fare downgrade.
In venv installi la versione che ti pare...

```
python3 -m pip install homeassistant==<versione>
```

Dopo provo in locale da me e ti dico come va.

Nel caso contattami su Telegram e vediamo insieme quando ho un attimo

**Reply by simille83** [reply]
Per caso hai risolto? Ho avuto lo stesso problema in seguito all'aggiornamento 2022.2.3 e anche la scheda energia che ha smesso di funzionare, nel mio log l'errore è legato a numpy, almeno credo.

Importing the numpy C-extensions failed. This error can happen for many reasons, often due to issues with your setup or how NumPy was https://numpy.org/devdocs/user/troubleshooting-importerror.html
The NumPy version is: "1.21.2"
Original error was: /lib/arm-linux-gnueabihf/libm.so.6: version `GLIBC_2.29' not found (required by /srv/homeassistant/lib/python3.9/site-packages/numpy/core/_multiarray_umath.cpython-39-arm-linux-gnueabihf.so).

**Reply by manny199** [reply]
@simille83 Ciao, non ho avuto ancora il tempo di dedicarmi al problema che dunque non ho risolto.
Riavvio sempre da riga di comando ...
Se hai sviluppi informami.
Grazie

**Reply by snakuzzo** [reply]
@simille83 hai cercato l'errore su Google ?

Leggi [QUI](https://github.com/numpy/numpy/issues/20371)

**Reply by simille83** [reply]
> @simille83 hai cercato l'errore su Google ?
> 
> Leggi [QUI](https://github.com/numpy/numpy/issues/20371)

Io sto eseguendo HA in venv con python 3.9 su raspian buster, quindi devo aggiornare quest'ultimo alla versione 11 in modo che non crei problema con la versione 3.9 di python?

**Reply by Scialla** [reply]
In sostanza... più lo sviluppo di HA andrà avanti e maggiori saranno le incompatibilità con le librerie supportate da Buster, sarà sempre peggio.
Per risolvere dovresti aggiornare il sistema a Bullseye e reinstallare HA ma, a sto punto (visto che ci sono appena passato), valuta di fare un lavoro che non ti darà problemi in futuro.
Io ho aggiornato a Bullseye ma avevo problemi che non sono riuscito a risolvere (non mi si aggiornavano alcune librerie) e ho deciso di ripartire da zero con Bullseye e usare docker. Con docker non avresti più questo tipo di problemi in quanto ti scarichi una immagine con già tutto il necessario (aggiornato) per il funzionamento di HA e non perderesti nulla della tua configurazione in quanto puoi backuppare e reimportare tutta la tua configurazione, c'è pure la guida apposita..
Io ho smesso di litigare (e in fondo un pò mi manca :D )

**Reply by simille83** [reply]
> In sostanza... più lo sviluppo di HA andrà avanti e maggiori saranno le incompatibilità con le librerie supportate da Buster, sarà sempre peggio. Per risolvere dovresti aggiornare il sistema a Bullseye e reinstallare HA ma, a sto punto (visto che ci sono appena passato), valuta di fare un lavoro che non ti darà problemi in futuro. Io ho aggiornato a Bullseye ma avevo problemi che non sono riuscito a risolvere (non mi si aggiornavano alcune librerie) e ho deciso di ripartire da zero con Bullseye e usare docker. Con docker non avresti più questo tipo di problemi in quanto ti scarichi una immagine con già tutto il necessario (aggiornato) per il funzionamento di HA e non perderesti nulla della tua configurazione in quanto puoi backuppare e reimportare tutta la tua configurazione, c'è pure la guida apposita.. Io ho smesso di litigare (e in fondo un pò mi manca :D )

Si in effettiè quello che vorrei fare anche io, mancano le conoscenze su docker e il tempo per farlo. Ma su un Raspberry 3 riesco a far girare tutto con docker? Principalmente HA e magari qualche altra istanza tipo PiHole. Perché altrimenti stavo anche valutando Home Assistant OS.

**Reply by Scialla** [reply]
Per Docker basta veramente poco, è moooolto più facile che in Venv; guardati la guida su inDomus per migrare da Venv a Docker e guardati la guida per installare HA con Docker-Compose, nulla di complicato.

Io uso un 3b con docker e HA, z2m, mosquitto, caddy, retropie.. basta e avanza

**Reply by manny199** [reply]
In un'altra sede, ho già installato Home Assistant su docker usando anche docker compose e vi assicuro che è eccezionale.
L'unica mia remora é se il supporto docker per home assistant ci sarà per sempre ...

**Reply by snakuzzo** [reply]
Stai scherzando, spero :D
Da dove è uscito fuori questo dubbio?

**Reply by snakuzzo** [reply]
Se hai questo dubbio, probabilmente non ti è chiaro cosa sia Docker

**Reply by snakuzzo** [reply]
Se ti stai facendo confondere dal discorso HA Supervised su docker su debian... quello è legato al supervisor e non ad ha in se.
Considera che hassos è un insieme di container docker. Se non ci fosse più docker, non esisterebbe nemmeno HA

**Reply by simille83** [reply]
> Per Docker basta veramente poco, è moooolto più facile che in Venv; guardati la guida su inDomus per migrare da Venv a Docker e guardati la guida per installare HA con Docker-Compose, nulla di complicato.
> 
> Io uso un 3b con docker e HA, z2m, mosquitto, caddy, retropie.. basta e avanza

Si la guida l'avevo già vista e letta, ultima cosa prima di installare docker e migrare HA conviene comunque aggiornare a Bullesey o si può fare anche in seguito?
Grazie ancora per i consigli.

**Reply by snakuzzo** [reply]
Secondo me fai prima a mettere il sistema operativo da zero.
Se aggiorni (soprattutto se sei su SD) ci mette un bel po' di tempo e comunque ti potresti scontrare con qualche cosa da sistemare.
Se hai solo Core...backup della dir di conf, rimetti l'os e parti da lì...vedrai che ci vuole veramente poco

**Comment by manny199** [comment]
Trovato questo errore ma ancora senza soluzione
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1183, in _sendfile_fallback
    read = await self.run_in_executor(None, file.readinto, view)
asyncio.exceptions.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 514, in start
    resp, reset = await task
  File "/srv/homeassistant/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 460, in _handle_request
    reset = await self.finish_response(request, resp, start_time)
  File "/srv/homeassistant/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 613, in finish_response
    await prepare_meth(request)
  File "/srv/homeassistant/lib/python3.9/site-packages/aiohttp/web_fileresponse.py", line 286, in prepare
    return await self._sendfile(request, fobj, offset, count)
  File "/srv/homeassistant/lib/python3.9/site-packages/aiohttp/web_fileresponse.py", line 99, in _sendfile
    await loop.sendfile(transport, fobj, offset, count)
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1162, in sendfile
    return await self._sendfile_fallback(transport, file,
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1192, in _sendfile_fallback
    await proto.restore()
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 263, in restore
    self._transport.resume_reading()
  File "/usr/local/lib/python3.9/asyncio/sslproto.py", line 343, in resume_reading
    self._ssl_protocol._transport.resume_reading()
AttributeError: 'NoneType' object has no attribute 'resume_reading'
```

**Reply by snakuzzo** [reply]
Segui [questa issue](https://github.com/home-assistant/core/issues/45641)
